client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -117,6 +119,16 @@ public void updateRetentionStreamCut(Map<Stream, StreamCut> streamCuts) {
                synchronizer.getState().getConfig().getRetentionType().toString());
     }
 
+    public UUID getGroupId() {
+        synchronizer.fetchUpdates();
```

Comment:
It's strange that this calls fetchUpdates. This is a blocking remote call. This is a getter so it's confusing. It is also not clear when the readerGroupId would change and what consistency is attempting to be enforced by updating first.

Reply:
Here we do not need to call fetchUpdates for consistency as the groupId never changes.